server/src/controllers/job_posts/getOneJobPost.js

```diff
@@ -0,0 +1,42 @@
+const { query } = require("express");
+const JobPosting = require("../../models/job_posting.model");
+const logger = require("pino")();
+/**
+a) GET request (get a job post’s information, get all columns based on id) using the job post id parameter
+Return all info about that specific job post
+ */
+
+const getJobPostsRequestHandler = async (req, res) => {
+    // check method is GET
+    if (req.method !== 'GET') {
+        return res.status(405).json({ message: 'Method not allowed, only GET methods allowed.' });
+    }
+
+    try{
+        // ------ Get Job Id
+        const jobPostId = req?.query?.id ? parseInt(req.query.id, 10) : null;
+ 
+        // Return to include id, title, employer, application_close_date, state
+        const requiredFields = ['id', 'title', 'employer', 'location', 'hours_per_week', 'rate_of_pay_min', 'rate_of_pay_max', 'rate_of_pay_frequency', 'job_type', 'close_date', 'job_description', 'custom_questions', 'creator', 'state'];
+
+        // Get one Job Posts
+        const jobPost = await JobPosting.findOne({where: {id: jobPostId}, attributes: requiredFields});
```

Comment:
Maybe a check here to see if there is an existing job post with the associated jobPostId might be good.

Reply:
Based on what I searched up, the object will return NULL and the response object will return NULL for the 'jobPost' key. If this is okay with you then feel free to ignore this comment!